cogs/annual_handover_and_reset.py

```diff
@@ -0,0 +1,208 @@
+
+__all__: Sequence[str] = ("HandoverCommandCog", "AnnualResetRolesCommandCog")
+
+import logging
+from logging import Logger
+from typing import Final
+
+import discord
+
+from db.core.models import GroupMadeMember
+from utils import AllChannelTypes, CommandChecks, TeXBotApplicationContext, TeXBotBaseCog
+
+logger: Logger = logging.getLogger("TeX-Bot")
+
+
+class HandoverCommandCog(TeXBotBaseCog):
+    """Cog class that defines the handover command."""
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="handover",
+        description="Initiates the annual Discord handover procedure for new committee",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def handover(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition & callback response of the "handover" command.
+
+        The "handover" command runs the relevant handover methods
+        which will perform the following actions:
+        - Give the "Committee" role to any users that have the "Committee-Elect" role
+        - Remove the "Committee-Elect" role from any user that has it
+        - Remove the permissions for the "Committee-Elect" role
+          from all channels except "#handover"
+
+        To do this, TeX-Bot will need to hold a role above that of the committee role.
+        """
+        # NOTE: Shortcut accessors are placed at the top of the function, so that the exceptions they raise are displayed before any further errors may be sent
+        main_guild: discord.Guild = self.bot.main_guild
+        committee_role: discord.Role = await self.bot.committee_role
+        committee_elect_role: discord.Role = await self.bot.committee_elect_role
+
+        initial_response: discord.Interaction | discord.WebhookMessage = await ctx.respond(
+            ":hourglass: Running handover procedures... :hourglass:",
+        )
+        logger.debug("Running the handover command!")
+
+        if main_guild.me.top_role.position < committee_role.position:
+            logger.debug(
+                "Handover command aborted because the bot did not "
+                "hold a role above the committee role.",
+            )
+            await initial_response.edit(
+                content=(
+                    ":warning: This command requires TeX-Bot to hold a role higher than "
+                    "that of the \"Committee\" role to perform this action. Operation aborted."
+                    " :warning:"
+                ),
+            )
+            return
+
+        category: discord.CategoryChannel
+        for category in main_guild.categories:
+            if "committee" not in category.name.lower() or "archive" in category.name.lower():
+                continue
+
+            await initial_response.edit(
+                content=f":hourglass: Updating channels in category: {category.name} "
+                ":hourglass:",
```

Comment:
Multi-line strings should be separated with brackets & response message can be improved.

Fix:
```python
await initial_reponse.edit(
    content=(
        f":hourglass: Updating channels in category: {category.name} "
        ":hourglass:"
    ),
)